sql/core/src/main/scala/org/apache/spark/sql/scripting/SqlScriptingInterpreter.scala

```diff
@@ -17,32 +17,42 @@
 
 package org.apache.spark.sql.scripting
 
-import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.{Row, SparkSession}
 import org.apache.spark.sql.catalyst.analysis.UnresolvedIdentifier
-import org.apache.spark.sql.catalyst.parser.{CompoundBody, CompoundPlanStatement, IfElseStatement, SingleStatement, WhileStatement}
-import org.apache.spark.sql.catalyst.plans.logical.{CreateVariable, DropVariable, LogicalPlan}
+import org.apache.spark.sql.catalyst.plans.logical.{CompoundBody, CompoundPlanStatement, CreateVariable, DropVariable, IfElseStatement, LogicalPlan, SingleStatement, WhileStatement}
 import org.apache.spark.sql.catalyst.trees.Origin
 
 /**
  * SQL scripting interpreter - builds SQL script execution plan.
+ *
+ * @param session
+ *   Spark session that SQL script is executed within.
  */
-case class SqlScriptingInterpreter() {
+case class SqlScriptingInterpreter(session: SparkSession) {
+
+  /**
+   * Execute the provided CompoundBody and return the result.
+   *
+   * @param compoundBody
+   *   CompoundBody to execute.
+   * @return Result of the execution.
+   */
+  def execute(compoundBody: CompoundBody): Seq[Row] = {
```

Comment:
I'm OK with calling `df.collect()` to eagerly execute every statement, in the first version. But I disagree with this API design that exposes `Seq[Row]`, which forces us to put all the data on the driver.

I'd rather return `Iterator[DataFrame]`, while each DataFrame is just `LocalRelation` which is the result of eagerly executed statement.

Reply:
The idea for execute is to return only the last `DataFrame`. We can also wrap it here in `LocalRelation`, but we decided to do it in the same place where `DataFrame` was created before -> in `SparkSession`.

Reply:
We need an agreement on this. cc @mitkedb Did you discussed this with Wenchen?